split-webapp/split/src/pages/SignUp.js

```diff
@@ -21,28 +21,53 @@ class SignUp extends Component {
         password: "",
         confPassword: ""
       },
-      failed: false
+      failed: false,
+      validation: {
+        validPassword: true,
+        matchPassword: true
+      }
     };
   }
 
   handleTextChange = event => {
-    const { user } = this.state;
+    const { user, validation } = this.state;
     user[event.target.name] = event.target.value;
-    this.setState({ user, failed: false });
+    validation.validPassword = true;
+    this.setState({ user, failed: false, validation });
+  };
+
+  handleBlurChange = () => {
+    const { validation } = this.state;
+    validation.matchPassword = this.validatePassword();
+    validation.validPassword = !this.emptyPassword();
+    this.setState({ validation });
   };
 
   // Creates a new user JSON and sends to the create account end point.
   // User should get logged in as well as session created.
 
   createUser() {
-    const { user } = this.state;
+    const { user, validation } = this.state;
     const { history } = this.props;
 
-    if (!(this.validateUsername() && this.validatePassword())) {
+    if (!this.validateUsername()) {
       this.setState({
         failed: true
       });
-    } else {
+    }
+
+    if (this.emptyPassword()) {
+      validation.validPassword = false;
+      this.setState({
+        validation
+      });
+    }
+
+    if (
+      this.validateUsername &&
+      !this.emptyPassword &&
```

Comment:
`validateUsername` and `emptyPassword` are functions, I assume you mean to check the result of the functions here?